src/test/java/com/squareup/kotlinpoet/FunSpecTest.kt

```diff
@@ -259,6 +259,22 @@ class FunSpecTest {
       |fun foo(nodoc: kotlin.Boolean): kotlin.String = "foo"""".trimMargin())
   }
 
+  @Test fun functionWithReturnKDocAndMainKdoc() {
+    val funSpec = FunSpec.builder("foo")
+        .addParameter("nodoc", Boolean::class)
+        .returns(String::class, kdoc = "the foo.")
+        .addCode("return %S", "foo")
+        .addKdoc("Do the foo")
+        .build()
+    assertThat(funSpec.toString()).isEqualTo("""
+      |/**
+      | * Do the foo
+      | *
+      | * @return the foo.
+      | */
+      |fun foo(nodoc: kotlin.Boolean): kotlin.String = "foo"""".trimMargin())
```

Comment:
Don't we want a new line before `"""`?

Reply:
Yes please

Reply:
Ouch, this snippet actually looks broken, notice the extra closing bracket at the end of the line... Normally you'd do `.addStatement("return %S", "foo")`, which would emit correct code. Not sure this is worth filing.

Reply:
Thanks! I went around and fixed that for some other tests also (written by yours truly, guilty as charged). 
0b9f23257bfa0ec49dac663c93b43ab3d128c83b